tests/cypress/support/apollo.ts

```diff
@@ -0,0 +1,64 @@
+import { ApolloClient, InMemoryCache, ApolloLink, from, NormalizedCacheObject } from '@apollo/client/core'
+import { setContext } from '@apollo/client/link/context'
+import { onError } from '@apollo/client/link/error'
+import { BatchHttpLink } from '@apollo/client/link/batch-http'
+
+interface authMethod {
+    token?: string
+    username?: string
+    password?: string
+    jsessionid?: string
+}
+
+interface httpHeaders {
+    authorization?: string
+    Cookie?: string
+}
+
+export const apolloClient = (authMethod?: authMethod, baseUrl?: string): ApolloClient<NormalizedCacheObject> => {
+    const testBaseUrl = baseUrl === undefined ? Cypress.config().baseUrl : baseUrl
+
+    const httpLink = new BatchHttpLink({
```

Comment:
Would this cause any flaky tests if the network is not stable?
Then if we are testing function or path base is this more E2E?

Reply:
yeah, it is more E2E.
We don't expect the network to be unstable, since it's all running on the same machine in docker containers.

Reply:
I assume each repo will have it's own docker container running their respective test?
Is there a way to separate the run of "integration" vs "E2E" test inside the docker container?

Reply:
the "E2E" are going to be ran slightly differently. "Integration" tests are co-located with the code they're testing and are ran as you see in this setup. "E2E" are going to be in their own repo, pulling/reusing various parts of the many "integration" tests. Therefore, "E2E" tests are always going to be separate from "Integration", simply because it's a separate repo running in a different configuration.